### ISSUE TITLE
Potential octal bugs

### ISSUE THREAD

**Issue by DavidHuber-NOAA** [NOAA-EMC/global-workflow#1195]
Your bug may already be reported!
Please search on the [Issue tracker](https://github.com/NOAA-EMC/global-workflow/issues) before creating one.

**Expected behavior**
Numbers starting with "0" should almost a always be considered decimals.

**Current behavior**
At least one job, [the earc job](https://github.com/NOAA-EMC/global-workflow/blob/8581eacf1a26efba614a1c04715fbc24cb7ae858/scripts/exgdas_enkf_earc.sh#L51), does not specify that a number should be considered a decimal when performing math on it (in this case, dd and mm should be considered decimals when calculating nday, which results in octal errors when mm or dd are 08 or 09).

A similar bug can be found in scripts/exgfs_atmos_post.sh, lines [277, 297, 303, 310, 391, etc](https://github.com/NOAA-EMC/global-workflow/blob/8581eacf1a26efba614a1c04715fbc24cb7ae858/scripts/exgfs_atmos_post.sh#L277).

**Machines affected**
All

**To Reproduce**

Run a gdasearc0[1-8] job for a date in August or September.

**Additional Information**

I went searching through the global workflow scripts for other instances where this may possibly occur.  Places of interest include 

ush/tropcy_relocate_extrkr.sh:424
ush/getges.sh:1342-1350; ghm* and ghp* are set to fhm* and fhp*, which could be 08 or 09.  Thus something like [[ $ghm6 -lt 100 ]] could cause issues.
ush/fv3gfs_downstream_nems.sh:242, 270
scripts/exgfs_pmgr.sh: 28, 32, 34, 47
scripts/exgfs_prdgen_manager.sh: 19, 27, 29, 32, 47 (if reinstated)
scripts/exglobal_atmos_pmgr.sh (as above)
~scripts/exgfs_atmos_postsnd.sh: 81, 88, 111~ Fixed by #1212
scripts/exgdas_atmos_nawips.sh: 72

**Possible Implementation**

**Comment by DavidHuber-NOAA** [comment]
Another bug is located at exgfs_atmos_post.sh, line 258.

**Comment by KateFriedman-NOAA** [comment]
> Another bug is located at exgfs_atmos_post.sh, line 258.

I also see the same error and other related ones in testing on Orion:
```
Orion-login-1[13] /work/noaa/stmp/kfriedma/comrot/cplmodule$ grep error logs/2013010100/gfs* | grep "value too great"
logs/2013010100/gfspost_f018-f018.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 018: value too great for base (error token is "018")
logs/2013010100/gfspost_f018-f018.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 018: value too great for base (error token is "018")
logs/2013010100/gfspost_f018-f018.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 018: value too great for base (error token is "018")
logs/2013010100/gfspost_f018-f018.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 018: value too great for base (error token is "018")
logs/2013010100/gfspost_f048-f048.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 048: value too great for base (error token is "048")
logs/2013010100/gfspost_f048-f048.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 048: value too great for base (error token is "048")
logs/2013010100/gfspost_f048-f048.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 048: value too great for base (error token is "048")
logs/2013010100/gfspost_f048-f048.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 048: value too great for base (error token is "048")
```
```
Orion-login-1[16] /work/noaa/stmp/kfriedma/comrot/cycmodule$ grep error logs/*/* | grep "value too great"
logs/2022010200/gfspost_f000-f009.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 009: value too great for base (error token is "009")
logs/2022010200/gfspost_f000-f009.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 009: value too great for base (error token is "009")
logs/2022010200/gfspost_f000-f009.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 009: value too great for base (error token is "009")
logs/2022010200/gfspost_f000-f009.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 009: value too great for base (error token is "009")
logs/2022010200/gfspost_f012-f021.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 018: value too great for base (error token is "018")
logs/2022010200/gfspost_f012-f021.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 018: value too great for base (error token is "018")
logs/2022010200/gfspost_f012-f021.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 018: value too great for base (error token is "018")
logs/2022010200/gfspost_f012-f021.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 018: value too great for base (error token is "018")
logs/2022010200/gfspost_f036-f042.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 039: value too great for base (error token is "039")
logs/2022010200/gfspost_f036-f042.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 039: value too great for base (error token is "039")
logs/2022010200/gfspost_f036-f042.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 039: value too great for base (error token is "039")
logs/2022010200/gfspost_f036-f042.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 039: value too great for base (error token is "039")
logs/2022010200/gfspost_f045-f051.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 048: value too great for base (error token is "048")
logs/2022010200/gfspost_f045-f051.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 048: value too great for base (error token is "048")
logs/2022010200/gfspost_f045-f051.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 048: value too great for base (error token is "048")
logs/2022010200/gfspost_f045-f051.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 048: value too great for base (error token is "048")
logs/2022010200/gfspost_f063-f069.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 069: value too great for base (error token is "069")
logs/2022010200/gfspost_f063-f069.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 069: value too great for base (error token is "069")
logs/2022010200/gfspost_f063-f069.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 069: value too great for base (error token is "069")
logs/2022010200/gfspost_f063-f069.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 069: value too great for base (error token is "069")
logs/2022010200/gfspost_f072-f078.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 078: value too great for base (error token is "078")
logs/2022010200/gfspost_f072-f078.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 078: value too great for base (error token is "078")
logs/2022010200/gfspost_f072-f078.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 078: value too great for base (error token is "078")
logs/2022010200/gfspost_f072-f078.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 078: value too great for base (error token is "078")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 081: value too great for base (error token is "081")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 081: value too great for base (error token is "081")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 081: value too great for base (error token is "081")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 081: value too great for base (error token is "081")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 084: value too great for base (error token is "084")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 084: value too great for base (error token is "084")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 084: value too great for base (error token is "084")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 084: value too great for base (error token is "084")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 087: value too great for base (error token is "087")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 087: value too great for base (error token is "087")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 087: value too great for base (error token is "087")
logs/2022010200/gfspost_f081-f087.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 087: value too great for base (error token is "087")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 090: value too great for base (error token is "090")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 090: value too great for base (error token is "090")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 090: value too great for base (error token is "090")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 090: value too great for base (error token is "090")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 093: value too great for base (error token is "093")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 093: value too great for base (error token is "093")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 093: value too great for base (error token is "093")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 093: value too great for base (error token is "093")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 096: value too great for base (error token is "096")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 096: value too great for base (error token is "096")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 096: value too great for base (error token is "096")
logs/2022010200/gfspost_f090-f096.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 096: value too great for base (error token is "096")
logs/2022010200/gfspost_f099-f105.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 258: ((: 099: value too great for base (error token is "099")
logs/2022010200/gfspost_f099-f105.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 278: ((: 099: value too great for base (error token is "099")
logs/2022010200/gfspost_f099-f105.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 284: ((: 099: value too great for base (error token is "099")
logs/2022010200/gfspost_f099-f105.log:/work/noaa/global/kfriedma/git/feature-module-order-clean/scripts/exgfs_atmos_post.sh: line 372: ((: 099: value too great for base (error token is "099")
```

**Comment by KateFriedman-NOAA** [comment]
The bugs noted in this issue will be resolved by work being done for issue #270.

**Comment by WalterKolczynski-NOAA** [comment]
More bugs were found by Judy in post. I'm changing it so fhr is never zero-padded and introducing fhr3 for the zero-padded version.
